### DISCUSSION TITLE
想要测试lvs的极限pps

### DISCUSSION THREAD

**Discussion by Sincere001** [baidu/dperf#463] [Q&A] [answered]
彭老师，您好，我想问一下
1.我想使用dperf测试lvs的极限pps时，是否只需要客户端就可以，下面是我当前的配置，只能 看见自己发包了，却没有收包，且抓包也没有发现，请问问题出在哪里呢。
mode        client
keepalive   60s
cpu         0
duration    10s
protocol    http
cps         100
cc          1000
port        0000:1a:00.2    39.135.97.81   39.135.97.1 74:4a:a4:20:1c:00 
client      39.135.97.81     10
server      39.135.97.80     1
listen      8080              1
2.如果问题1的原因是因为没有配置server的原因导致的，那么我现在的交换机与两台lvs机器是通过ospf协议，是否需要在两台lvs上都配置server呢

**Comment by pengjianzhang** [comment]
需要dperf client，dperf server（作为LVS 的rs）
dperf server可以作为为两个LVS的rs
建议先跑通一些简单场景：
-  case1: dperf client -->dperf server
-  case2: dperf client -->1 LVS ---> dperf server
- case 3: dperf client---> 2 LVS ---> dperf server

**Reply by Sincere001** [reply]
在启动服务端配置时，遇到rte_eal_init fail dpdk_eal_init fail dpdk_init fail。请问该怎么解决，搜到的资料没有具体说明解决办法

**Reply by Sincere001** [reply]
您好，我在启动客户端和服务端时均报错，想问下您该怎么解决
报错信息如下
EAL: Could not find space for memseg. Please increase CONFIG_RTE_MAX_MEMSEG_PER_TYPE and/or CONFIG_RTE_MAX_MEM_PER_TYPE in configuration.
EAL: Couldn't remap hugepage files into memseg lists
EAL: FATAL: Cannot init memory
EAL: Cannot init memory
rte_eal_init fail
dpdk_eal_init fail
dpdk init fail

client端配置
mode        client
cpu         0
cc          50
cps         50
duration    60s
tx_burst    1024
protocol    tcp
keepalive   10us
port        0000:81:00.1     39.135.131.135 39.135.131.129   04:b0:e7:f7:44:2c
client      39.135.131.135      2
server      39.135.131.134      1
listen      80              1

server端配置
mode        server
cpu         0
duration    60s
tx_burst    1024
protocol    tcp
keepalive   10us
port        0000:81:00.1     39.135.131.135 39.135.131.129   04:b0:e7:f7:44:2c
client      39.135.131.155      2
server      39.135.131.130      1
listen      80              1

**Reply by yuexiaocai** [reply]
> 您好，我在启动客户端和服务端时均报错，想问下您该怎么解决 报错信息如下 EAL: Could not find space for memseg. Please increase CONFIG_RTE_MAX_MEMSEG_PER_TYPE and/or CONFIG_RTE_MAX_MEM_PER_TYPE in configuration. EAL: Couldn't remap hugepage files into memseg lists EAL: FATAL: Cannot init memory EAL: Cannot init memory rte_eal_init fail dpdk_eal_init fail dpdk init fail
> 
> client端配置 mode client cpu 0 cc 50 cps 50 duration 60s tx_burst 1024 protocol tcp keepalive 10us port 0000:81:00.1 39.135.131.135 39.135.131.129 04:b0:e7:f7:44:2c client 39.135.131.135 2 server 39.135.131.134 1 listen 80 1
> 
> server端配置 mode server cpu 0 duration 60s tx_burst 1024 protocol tcp keepalive 10us port 0000:81:00.1 39.135.131.135 39.135.131.129 04:b0:e7:f7:44:2c client 39.135.131.155 2 server 39.135.131.130 1 listen 80 1

你好，我也遇到了类似的问题，请问后来是怎么解决的呢？

**Reply by pengjianzhang** [reply]
是不是没有配大页，参考dperf 相关文档 https://dperf.org/

Set up DPDK
Set up dperf

**Reply by Sincere001** [reply]
> > 您好，我在启动客户端和服务端时均报错，想问下您该怎么解决 报错信息如下 EAL: Could not find space for memseg. Please increase CONFIG_RTE_MAX_MEMSEG_PER_TYPE and/or CONFIG_RTE_MAX_MEM_PER_TYPE in configuration. EAL: Couldn't remap hugepage files into memseg lists EAL: FATAL: Cannot init memory EAL: Cannot init memory rte_eal_init fail dpdk_eal_init fail dpdk init fail
> > client端配置 mode client cpu 0 cc 50 cps 50 duration 60s tx_burst 1024 protocol tcp keepalive 10us port 0000:81:00.1 39.135.131.135 39.135.131.129 04:b0:e7:f7:44:2c client 39.135.131.135 2 server 39.135.131.134 1 listen 80 1
> > server端配置 mode server cpu 0 duration 60s tx_burst 1024 protocol tcp keepalive 10us port 0000:81:00.1 39.135.131.135 39.135.131.129 04:b0:e7:f7:44:2c client 39.135.131.155 2 server 39.135.131.130 1 listen 80 1
> 
> 你好，我也遇到了类似的问题，请问后来是怎么解决的呢？

我当时是大页设置的很大80G出的这个问题，改成了4G解决的

**Reply by yuexiaocai** [reply]
> 是不是没有配大页，参考dperf 相关文档 https://dperf.org/
> 
> Set up DPDK Set up dperf

@pengjianzhang 大页配置了的，起初是设置了140个1GB大页，刚刚按照 @Sincere001 说的，尝试改成了8个1GB大页，就能够正常运行了。问题与这个类似：https://mails.dpdk.org/archives/dev/2023-May/268032.html

我注意到 `EAL: Couldn't remap hugepage files into memseg lists` 这个报错是在 `eal_legacy_hugepage_init()` 函数中产生的，也就是说进入了 Legacy Memory Mode，但我并未指定 `--legacy-mem` 的eal 参数，正常来说应该是 Dynamic Memory Mode 才对。

**Reply by pengjianzhang** [reply]
@yuexiaocai   你用的最新代码吗？ 1.7可能强制--legacy-mem了， 最新的代码应该去掉了--legacy-mem

**Reply by yuexiaocai** [reply]
> @yuexiaocai 你用的最新代码吗？ 1.7可能强制--legacy-mem了， 最新的代码应该去掉了--legacy-mem

我用的 release 1.7，那估计就是这里问题了，谢谢彭老师的解答！

**Reply by pengjianzhang** [reply]
@yuexiaocai 请测一下最新代码，是否还有这个问题，谢谢

**Reply by yuexiaocai** [reply, marked as answer]
> @yuexiaocai 请测一下最新代码，是否还有这个问题，谢谢

@pengjianzhang 测试了main分支的代码，128个1GB大页也可以正常运行了，谢谢老师

**Comment by pengjianzhang** [comment]
轻阅读 dperf 相关文档 https://dperf.org/

Set up DPDK
Set up dperf
Performance Testing

**Reply by Sincere001** [reply]
好的，谢谢彭老师，现在问题已经解决了。服务端和客户端都可以正常启动了，但是服务端收不到客户端的流量，客户端那边的连接全部都skERR重传出错3次。我反复检查了配置文件没有问题，想问下您是什么原因

**Reply by Sincere001** [reply]
我的bond0网卡是用的4个10G的网卡，绑定的时候需要都绑定上么

**Comment by pengjianzhang** [comment]
dperf 不要用Bond

**Reply by Sincere001** [reply]
彭老师，我现在client（135）到server（134）端能够抓包发现，一直在重传，互相收不到ack的包，134向135、135向134发都是这样，想问下您该如何解决，server进程无论启动与否都是这样

**Comment by Sincere001** [comment]
![4c6ebff7be24ed5eff11539746b98c9b_16760](https://github.com/user-attachments/assets/e9e40bc4-e1b2-4d53-a572-e1c38c57786c)
想请问下这边client启动后 cpuusage直接100 是什么原因。配置如下

mode client
cpu 2
duration        2m
payload_size    1
cps             10
port            xxxx
client          xxxx  1
server          xxxx  1
listen          80              1

**Reply by pengjianzhang** [reply]
1. 什么网卡
2. 最新代码是否能复现

**Comment by Sincere001** [comment]
网卡是Intel Corporation 82599ES
![18281e58e7c5bdf2b8b739cc152225e5_54824](https://github.com/user-attachments/assets/96556d06-8775-4563-a849-af4b2cda0ff5)
shi使用main分支的代码仍存在这个问题

**Reply by pengjianzhang** [reply]
看起来是第0秒cpu显示100%, 后面是不是正常？

**Reply by Sincere001** [reply]
是的，后面是正常的。我现在是cli到ser能够正常接收到流量。中间加上了lvs之后，再测试过程中，lvs会把后面的rs设备软下线，想知道这种该怎么解决

**Reply by Sincere001** [reply]
我看原因是，ser那里上来cpu100导致的，lvs探不到后面rs了

**Reply by pengjianzhang** [reply]
server 配置里的'client'要增加lvs的地址
用的什么转发模式？

**Reply by Sincere001** [reply]
lvs是DR，server里面中的client添加了lvs的vip

**Reply by pengjianzhang** [reply]
建议测试其他模式，dperf会响应ARP，不适合DR模式

**Reply by pengjianzhang** [reply]
我考虑一下 看看能不能支持DR模式，给我发个邮件pengjianzhang@gmail.com ，我们会议里交流一下

**Reply by pengjianzhang** [reply]
请试一下这个分支

https://github.com/pengjianzhang/dperf/tree/arp_ignore

server上如下配置：


neigh_ignore
port  PCI  rsIP  gatewayIP
server vip 1
